src/kOS/Suffixed/ActiveResourceValue.cs

```diff
@@ -7,28 +7,55 @@ namespace kOS.Suffixed
+        private double capacity;
 
-        public ActiveResourceValue(Vessel.ActiveResource activeResource, SharedObjects shared)
+        public ActiveResourceValue(Vessel ves, int resId, SharedObjects shared)
         {
-            this.activeResource = activeResource;
+            vessel = ves;
+            resourceId = resId;
             this.shared = shared;
             InitializeActiveResourceSuffixes();
         }
 
         private void InitializeActiveResourceSuffixes()
         {
-            AddSuffix("NAME", new Suffix<StringValue>(() => activeResource.info.name, "The name of the resource (eg LiguidFuel, ElectricCharge)"));
-            AddSuffix("AMOUNT", new Suffix<ScalarValue>(() => activeResource.amount, "The resources currently available"));
-            AddSuffix("CAPACITY", new Suffix<ScalarValue>(() => activeResource.maxAmount, "The total storage capacity currently available"));
-            AddSuffix("DENSITY", new Suffix<ScalarValue>(() => activeResource.info.density, "The density of this resource"));
-            AddSuffix("PARTS", new Suffix<ListValue<PartValue>>(() => PartValueFactory.ConstructGeneric(activeResource.parts, shared), "The containers for this resource"));
+            AddSuffix("NAME", new Suffix<StringValue>(GetName, "The name of the resource (eg LiguidFuel, ElectricCharge)"));
+            AddSuffix("AMOUNT", new Suffix<ScalarValue>(GetAmount, "The resources currently available"));
+            AddSuffix("CAPACITY", new Suffix<ScalarValue>(GetCapacity, "The total storage capacity currently available"));
+            AddSuffix("DENSITY", new Suffix<ScalarValue>(GetDensity, "The density of this resource"));
+            // TODO: IMPORTANT fix this before release, though it isn't a documented suffix and there may be value to unifying the resource classes
+            //AddSuffix("PARTS", new Suffix<ListValue<PartValue>>(() => PartValueFactory.ConstructGeneric(activeResource.parts, shared), "The containers for this resource"));
         }
 
         public override string ToString()
         {
-            return string.Format("ACTIVERESOURCE({0},{1},{2})", activeResource.info.name, activeResource.amount, activeResource.maxAmount);
+            return string.Format("ACTIVERESOURCE({0},{1},{2})", GetName(), GetAmount(), GetCapacity());
+        }
+
+        public StringValue GetName()
+        {
+            return PartResourceLibrary.Instance.resourceDefinitions[resourceId].name;
+        }
+
+        public ScalarValue GetAmount()
+        {
+            vessel.GetConnectedResourceTotals(resourceId, out amount, out capacity, true);
+            return amount;
+        }
+
+        public ScalarValue GetCapacity()
+        {
+            vessel.GetConnectedResourceTotals(resourceId, out amount, out capacity, true);
+            return capacity;
+        }
+
+        public ScalarValue GetDensity()
+        {
+            return PartResourceLibrary.Instance.resourceDefinitions[resourceId].density;
```

Comment:
So if I understand the gist of this change: you're getting the values live on the fly now from the vessel instead of from buffered data in the activeResource object.  I assume this fixes a problem with the activeResource data being stale in some way?